core/domain/suggestion_services.py

```diff
@@ -599,6 +624,193 @@ def get_translation_suggestions_waiting_longest_for_review_per_lang(
+
+    Returns:
+        list(list(ReviewableSuggestionEmailInfo)). A list of suggestion
+        email content info objects for each reviewer. Each suggestion email
+        content info object contains the type of the suggestion, the language
+        of the suggestion, the suggestion content (question/translation) and
+        the date that the suggestion was submitted for review. For each user
+        the suggestion email content info objects are sorted in descending order
+        based on review wait time.
+    """
+    # Get each reviewer's review permissions.
+    users_contribution_rights = user_services.get_users_contribution_rights(
+        reviewer_ids
+    )
+
+    # Get the question suggestions that have been waiting longest for review.
+    question_suggestions = (
+        get_question_suggestions_waiting_longest_for_review()
+    )
+
+    # Create a dictionary to keep track of the translation suggestions that
+    # have been waiting longest for review for each language code.
+    translation_suggestions_by_lang_code_dict = {}
+
+    reviewers_reviewable_suggestion_infos = []
+
+    for user_contribution_rights in users_contribution_rights:
+        # Use a min heap because then the suggestions that have been waiting the
+        # longest for review (earliest review submission date) are automatically
+        # efficiently sorted.
+        suggestions_waiting_longest_heap = []
+        if user_contribution_rights.can_review_questions:
+            for question_suggestion in question_suggestions:
+                # Break early because we only want the top
+                # MAX_NUMER_OF_SUGGESTIONS_PER_REVIEWER number of suggestions.
+                if len(suggestions_waiting_longest_heap) == (
+                        MAX_NUMER_OF_SUGGESTIONS_PER_REVIEWER):
+                    break
+                # We can't include suggestions that were authored by the
+                # reviewer because reviewers aren't allowed to review their own
+                # suggestions.
+                elif question_suggestion.author_id != (
+                        user_contribution_rights.id):
+                    heapq.heappush(suggestions_waiting_longest_heap, (
+                        question_suggestion.last_updated, question_suggestion))
+
+        if user_contribution_rights.can_review_translation_for_language_codes:
+            for language_code in (
+                    user_contribution_rights
+                    .can_review_translation_for_language_codes):
+                # Get a list of the translation suggestions in the language code
+                # from the datastore if we haven't already gotten them.
+                if language_code not in (
+                        translation_suggestions_by_lang_code_dict):
+                    translation_suggestions_by_lang_code_dict[language_code] = (
+                        get_translation_suggestions_waiting_longest_for_review_per_lang( # pylint: disable=line-too-long
```

Comment:
You probably don't need "per_lang" in the name of the function.

Reply:
Looks resolved.